XYZDrivers/src/main/java/com/xyzdrivers/models/jdbcDriver.java

```diff
@@ -0,0 +1,435 @@
+    {
+        results = DBMetaData.getPrimaryKeys("", "", table);
+        
+        if (!results.next())
+            throw new IllegalArgumentException("Could not find PRIMARY KEY in "+table+". Please specify the primaryKeyColumn.");
+        
+        return results.getString("PK_NAME");
+    }
+//exists
+    /**
+     * Checks if <code>table</code> exists.
+     * 
+     * @param table The name of the table to look for.
+     * 
+     * @return The boolean result of <code>(query results).next()</code>
+     * 
+     * @throws SQLException 
+     */
+    public boolean exists(String table)
+            throws SQLException
+    {
+        //prepare statement
+        statement = DB.prepareStatement("SELECT * FROM "+table);
+        //execute statement
+        results = statement.executeQuery();
+        //return results
+        return results.next();
+    }
+        
+    /**
+     * Checks if <code>column</code> exists in <code>table</code>.
+     * 
+     * @param table The name of the table containing the data to retrieve data from
+     * @param column The name of the column to look for in <code>table</code>
+     * 
+     * @return The boolean result of <code>(query results).next()</code>
+     * 
+     * @throws SQLException 
+     */
+    public boolean exists(String table, String column)
+            throws SQLException
+    {
+        //prepare statement
+        statement = DB.prepareStatement("SELECT "+column+" FROM "+table+" WHERE "+column+" = ?");
+        //execute statement
+        results = statement.executeQuery();
+        //return results
+        return results.next();
+    }
+    
+    /**
+     * Checks if <code>query</code> exists in <code>table->column</code>.
+     * 
+     * @param table The name of the table containing the item
+     * @param column The name of the column containing the item to retrieve from table
+     * @param item The item to be look for in <code>table->column</code>
+     * 
+     * @return The boolean result of <code>(query results).next()</code>
+     * 
+     * @throws SQLException 
+     */
+    public boolean exists(String table, String column, Object item)
+            throws SQLException
+    {
+        //prepare statement
+        statement = DB.prepareStatement("SELECT "+column+" FROM "+table+" WHERE "+column+" = ?");
+        statement.setObject(1, item);
+        //execute statement
+        results = statement.executeQuery();
+        //return results
+        return results.next();
+    }
+//retrieve
+    /**
+     * 
+     * @param table
+     * @return
+     * @throws SQLException 
+     */
+    public List<Object[]> retrieve(String table)
+            throws SQLException
+    {
+        List<Object[]> data;
+        Object[] column;
+        
+        //check table and table->column exist
+        if (!exists(table))
+            throw new IllegalArgumentException();
+        //prepare statement
+        statement = DB.prepareStatement("SELECT * FROM "+table);
```

Comment:
This is a repeating pattern and can be moved to a function:
```
statement = DB.prepareStatement(...);
results = statement.executeQuery();
```

e.g. (I haven't tried to compile this, so pinch of salt)
```
Object executeStatement(String sql, Object... parameters) throws SQLException {
    statement = DB.prepareStatement(sql);
    
    for (int i = 0; i < parameters.length; i++)
        statement.setObject(i + 1, parameters[i]);
    
    return statement.executeQuery();
}

results = executeStatement([SQL here], "parameter1", "parameter2", 3, 4); 
```